Extend Quirk shouldDisableContentChangeObserver() to the full YouTube.com

karlcow opened the pull request:
#### 3aa4d2baaa6ffe357686d195531347c47cc31039
<pre>
Extend Quirk shouldDisableContentChangeObserver() to the full YouTube.com
<a href="https://bugs.webkit.org/show_bug.cgi?id=263789">https://bugs.webkit.org/show_bug.cgi?id=263789</a>
rdar://115566212

Reviewed by Chris Dumez.

Video thumbnails on the YouTube site are not navigating to the Video
page when tapping them. This problem was fixed in the past for the
search and results page. This PR extends the quirks to the full YouTube
domain.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldDisableContentChangeObserver const):

Canonical link: <a href="https://commits.webkit.org/269888@main">https://commits.webkit.org/269888@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f73d7c771b404d10afa51f04a342380d052fe0f5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/23943 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/2058 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25032 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26085 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22056 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/24216 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3692 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24427 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22552 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24188 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1605 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20689 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26674 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1359 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21603 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27834 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21836 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21886 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25629 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1288 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18968 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/1303 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5729 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1699 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1633 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -198,8 +198,7 @@ bool Quirks::shouldDisableContentChangeObserver() const
 
-    auto& topDocument = m_document->topDocument();
-    if (isDomain("youtube.com"_s) && (topDocument.url().path().startsWithIgnoringASCIICase("/results"_s) || topDocument.url().path().startsWithIgnoringASCIICase("/watch"_s)))
+    if (isDomain("youtube.com"_s))
```
It should return be:
```
return isDomain("youtube.com"_s);
```

karlcow replied:
done. Thanks for the review. 
I also added the missing comment as a reference.

cdumez reviewed (approved):
LGTM with fix.

cdumez commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -193,16 +193,14 @@ bool Quirks::hasBrokenEncryptedMediaAPISupportQuirk() const
-        return true;
-
-    return false;
+    return (isDomain("youtube.com"_s);
```
extra parenthesis

karlcow replied:
🤦

karlcow replied:
fixed.

karlcow committed:
Extend Quirk shouldDisableContentChangeObserver() to the full YouTube.com
https://bugs.webkit.org/show_bug.cgi?id=263789
rdar://115566212

Reviewed by Chris Dumez.

Video thumbnails on the YouTube site are not navigating to the Video
page when tapping them. This problem was fixed in the past for the
search and results page. This PR extends the quirks to the full YouTube
domain.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldDisableContentChangeObserver const):

Canonical link: https://commits.webkit.org/269888@main
